smaug/core/network_builder.cpp

```diff
@@ -333,14 +335,15 @@ static Network* createNetworkFromProto(const GraphProto& graphProto,
 
     // Now every operator has been added into the network, we can connect them
     // together by adding edges in the graph view of the network.
-    for (int i = 0; i < graphProto.nodes_size(); i++) {
-        const NodeProto& node = graphProto.nodes(i);
+    for (const NodeProto& node : graphProto.nodes()) {
         Operator* op = network->getOperator(node.name());
         for (int i = 0; i < node.parents_size(); i++) {
-            std::string inputOpName = node.parents(i);
             int srcTensorIdx = node.src_tensors_indices(i);
-            Operator* inputOp = network->getOperator(inputOpName);
-            network->addEdge(inputOp, op, { srcTensorIdx, i });
+            if (srcTensorIdx != -1) {
```

Comment:
What does -1 mean again? Can we make this an enum or other named constant so the code is self explanatory?

Reply:
Here we assumed the number of parents should be equal to the number of input tensors. Now we can have an input tensor that doesn't come from an operator, so this "N/A" and "-1" pair indicates this case.